components/ProfilePage/SmartWalletContents.tsx

```diff
@@ -9,13 +10,20 @@ import ProfileToken from "./ProfileToken"
 import { useUserProvider } from "../../providers/UserProvider"
 import getIpfsLink from "../../lib/getIpfsLink"
 import { ItemTypes } from "./ItemTypes"
+import { transferCre8orToSmartWallet } from "../../lib/cre8or"
+import { useWalletCollectionProvider } from "../../providers/WalletCollectionProvider"
+import { useEthersSigner } from "../../hooks/useEthersSigner"
 
 const SmartWalletContents = () => {
   const { isHiddenEditable } = useProfileProvider()
   const { metaData, cre8orNumber } = useUserProvider()
+  const { toggleProfileFormattedCollection } = useWalletCollectionProvider()
+
   const [ownedNfts, setOwnedNfts] = useState([])
   const [hasSmartWallet, setHasSmartWallet] = useState(true)
   const provider = useMemo(() => getDefaultProvider(process.env.NEXT_PUBLIC_TESTNET ? 5 : 1), [])
+  const { address } = useAccount()
+  const signer = useEthersSigner()
```

Comment:
these should live in a hook 🪝

Reply:
personal preference: don't put `signer` in React components. 

create a modular `hook`. 🪝

Reply:
in this case `useERC721Transfer` hook 🪝

Reply:
made hook! 👍